packages/experimental-package/src/components/RemoveModal/RemovalModal.js

```diff
@@ -0,0 +1,69 @@
+import React, { useState } from 'react';
+import { Modal, TextInput } from 'carbon-components-react';
+import PropTypes from 'prop-types';
+
+export const RemovalModal = ({
+  body,
+  className,
+  header,
+  onRequestClose,
+  onRequestSubmit,
+  open,
+  primaryButtonText,
+  resource,
+  secondaryButtonText,
+  subheader,
+  textConfirmation,
+}) => {
+  const [userInput, setUserInput] = useState('');
+  const onChangeHandler = (e) => {
+    setUserInput(e.target.value);
+  };
+  const primaryButtonDisabled = textConfirmation && userInput !== resource;
+  return (
+    <Modal
+      danger
+      className={`removal-modal ${className}`}
+      open={open}
+      modalHeading={header}
+      modalLabel={subheader}
+      primaryButtonText={primaryButtonText}
+      primaryButtonDisabled={primaryButtonDisabled}
+      secondaryButtonText={secondaryButtonText}
+      onRequestSubmit={onRequestSubmit}
+      onRequestClose={onRequestClose}>
+      <p className="removal-modal-body">{body}</p>
+      {textConfirmation && (
+        <TextInput
+          id="remove-confirmation"
+          invalidText="A valid value is required"
```

Comment:
There should be no hardcoded English text in a component, except possibly for default values.